### ISSUE TITLE
'Internal Server Error' when fetching myValliant data

### ISSUE THREAD

**Issue by fodorg01** [signalkraft/mypyllant-component#117]
### Before submitting a new issue

- [x] I [redownloaded the latest release](https://my.home-assistant.io/redirect/hacs_repository/?owner=signalkraft&repository=mypyllant-component) and restarted Home Assistant
- [x] I've checked [known issues](https://signalkraft.com/mypyllant-component/#known-issues)
- [x] I've [enabled debug logs](https://signalkraft.com/mypyllant-component/3-contributing/#debugging) and attached the relevant logs to this ticket

### Problem description

No data is fetched from the Vaillant server. The app is however still working, that's why it is suspicious that it might be a problem with the integration.

### Logs

```
Error fetching myVAILLANT data: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/a8ad6dba-c13c-4a66-b7b9-99661bfcbdbf/currentSystem')
```

**Comment by fodorg01** [comment]
I have edited the config in order to enable the debug logs, and restarted Home Assistant again. After this restart the automation started working again.
Can we keep this open a little bit longer to see how frequently this problem occurs? If it occurs again, I can attach the debug logs.

**Comment by fodorg01** [comment]
Here is one log:
```
Logger: homeassistant.components.script.heating_set_temperature
Source: helpers/script.py:485
Integration: Script (documentation, issues)
First occurred: 11:25:13 AM (1 occurrences)
Last logged: 11:25:13 AM

Heating: Set temperature: Error executing script. Unexpected error for call_service at pos 1: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/vrc700/v1/systems/a8ad6dba-c13c-4a66-b7b9-99661bfcbdbf/zone/0/heating/quick-veto')
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 485, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 723, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 685, in _async_run_long_action
    return long_task.result()
           ^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2279, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2316, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 892, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 962, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 835, in async_service_temperature_set
    await entity.async_set_temperature(**kwargs)
  File "/config/custom_components/mypyllant/climate.py", line 563, in async_set_temperature
    await self.set_quick_veto(temperature=temperature)
  File "/config/custom_components/mypyllant/climate.py", line 436, in set_quick_veto
    await self.coordinator.api.quick_veto_zone_temperature(
  File "/usr/local/lib/python3.12/site-packages/myPyllant/api.py", line 533, in quick_veto_zone_temperature
    await self.aiohttp_session.patch(
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 691, in _request
    await raise_for_status(resp)
  File "/usr/local/lib/python3.12/site-packages/myPyllant/http_client.py", line 50, in on_raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 1060, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 500, message='Internal Server Error', url=URL('https://api.vaillant-group.com/service-connected-control/vrc700/v1/systems/a8ad6dba-c13c-4a66-b7b9-99661bfcbdbf/zone/0/heating/quick-veto')

```

and another:

```
This error originated from a custom integration.

Logger: custom_components.mypyllant.coordinator
Source: helpers/update_coordinator.py:345
Integration: myVAILLANT (documentation, issues)
First occurred: 11:49:18 AM (2 occurrences)
Last logged: 11:49:46 AM

Error fetching myVAILLANT data: 403, message='Forbidden', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/emf/v2/a8ad6dba-c13c-4a66-b7b9-99661bfcbdbf/currentSystem')
Error fetching myVAILLANT data: 403, message='Forbidden', url=URL('https://api.vaillant-group.com/service-connected-control/end-user-app-api/v1/systems/a8ad6dba-c13c-4a66-b7b9-99661bfcbdbf/meta-info/time-zone')
```

**Comment by fodorg01** [comment]
I can also observe that an actuation event (setting the temperature on the heating or on the hot water) doesn't always succeed. After calling the corresponding service, the value is reported to be changed, but only for a few seconds, then it goes back to the previous value. In my case my automation re-tries, but then it fails again... it then results in lots of re-tries, which is for sure not really healthy. Here is a diagram:

![image](https://github.com/signalkraft/mypyllant-component/assets/145673205/89f6c132-d6ff-44b4-a0f0-78ca3ccc6597)

Here my automation tries to set the heating temperature to 22,4 and then to 22,3 °C, but it is always immediately returning back to 22,5 (so my automation re-tries). This state can persist for 10-20 minutes, when the request then finally succeeds. 

![image](https://github.com/signalkraft/mypyllant-component/assets/145673205/90c345a7-dc5a-4df8-b37c-4e73572ea375)

Here is another automation, where it tries to set the hot water temperature only once, at 4:30 pm:

![image](https://github.com/signalkraft/mypyllant-component/assets/145673205/c2d0a991-a03e-480d-acd6-d840a148be82)

It is interesting to see that it changes the value, but then it falls back immediately. But at the moment I looked at this diagram the first time, it then finally made the change. (This automation doesn't do re-tries. So it seems like it really took 40 minutes till it succeeded.)

It will try downgrading to a previous version (I am on the latest b8).

**Comment by fodorg01** [comment]
Update: the app shows the same behaviour. E.g. I cannot change the heating mode to Auto, same for warm water.
So at the end this might be a problem with the service / server.
